billmategateway/controllers/front/accept.php

```diff
@@ -60,297 +60,297 @@ public function getmoduleId($method)
-							'&id_order=' . (int)$order_id);
-						die;
-					}
-				} else {
-
-
-                    if ($paymentInfo['PaymentData']['method'] == '1' OR '2' == $paymentInfo['PaymentData']['method']) {
-                        /** Paid with invoice */
-                        $this->paymentClass = 'invoice';
-                        $class_file = _PS_MODULE_DIR_.'billmategateway/methods/'.Tools::ucfirst($this->paymentClass).'.php';
-                        require_once($class_file);
-                        $class        = "BillmateMethod".Tools::ucfirst($this->paymentClass);
-                        $this->module = new $class;
-                    }
+                            $status = ($data['status'] == 'Cancelled') ? Configuration::get('PS_OS_CANCELED') : $status;
+                            $orderHistory->id_order = (int)$orderObject->id;
+                            $orderHistory->changeIdOrderState($status, (int)$orderObject->id, true);
+                            $orderHistory->add();
+                        }
+                        if (isset($this->context->cookie->billmatepno))
+                            unset($this->context->cookie->billmatepno);
+
+                        if ('' != Common::getCartCheckoutHash()) {
+                            $hash = Common::getCartCheckoutHash();
+                            Common::unsetCartCheckoutHash();
+                            $url = $this->context->link->getModuleLink(
+                                'billmategateway',
+                                'thankyou',
+                                array('billmate_hash' => $hash));
+                            Tools::redirectLink($url);
+                            die;
+                        } else {
 
 
-                    $return = array();
-                    $paymentInfo = $this->billmate->getPaymentinfo(array('number' => $data['number']));
-                    $result = $paymentInfo;
+                            Tools::redirectLink(__PS_BASE_URI__ . 'order-confirmation.php?key=' . $customer->secure_key .
+                                '&id_cart=' . (int)$this->context->cart->id . '&id_module=' . (int)$this->getmoduleId('billmate' . $this->method) .
+                                '&id_order=' . (int)$order_id);
+                            die;
+                        }
+                    } else {
 
-                    if (!isset($result['code']) && (isset($result['PaymentData']['order']['number']) && is_numeric($result['PaymentData']['order']['number']) && $result['PaymentData']['order']['number'] > 0)) {
 
-                    $status = ($this->method == 'invoice') ? Configuration::get('BINVOICE_ORDER_STATUS') : Configuration::get('BPARTPAY_ORDER_STATUS');
-                    $status = ($this->method == 'invoiceservice') ? Configuration::get('BINVOICESERVICE_ORDER_STATUS') : $status;
+                        if ($paymentInfo['PaymentData']['method'] == '1' OR '2' == $paymentInfo['PaymentData']['method']) {
```

Comment:
Also avoid Yoda conditions, should be: 

`if ($paymentInfo['PaymentData']['method'] == '1' OR $paymentInfo['PaymentData']['method'] == '2') {`